kore/src/Kore/Exec.hs

```diff
@@ -575,11 +575,11 @@ initializeProver definitionModule specModule within =
             specClaims =
                 map (Bifunctor.second fromMaybeChanged) changedSpecClaims
 
+        assertSomeClaims specClaims
```

Comment:
Could you add a small comment why this check comes first? It seems like the sort of thing which is obvious now, but will quickly be forgotten.